runtime/lua/vim/treesitter/query.lua

```diff
@@ -138,6 +138,13 @@ function M.get_query(lang, query_name)
   end
 end
 
+local query_cache = setmetatable({}, {
+  __index = function(tbl, key)
+    rawset(tbl, key, {})
+    return rawget(tbl, key)
+  end
+})
```

Comment:
is there already an abstraction  over this: `defaultdict({})`?

Reply:
Not that I know of, but we use the same pattern in other places (e.g. diagnostics), so having a standardized implementation would maybe be a good idea